ehcache-impl/src/test/java/org/ehcache/impl/internal/store/heap/bytesized/OnHeapStoreByteStatisticTest.java

```diff
@@ -0,0 +1,106 @@
+package org.ehcache.impl.internal.store.heap.bytesized;
+
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ExpiryPolicyBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.ehcache.core.internal.statistics.DefaultStatisticsService;
+import org.ehcache.core.internal.statistics.DefaultTierStatistics;
+import org.ehcache.spi.resilience.StoreAccessException;
+import org.junit.Test;
+
+import java.time.Duration;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+
+public class OnHeapStoreByteStatisticTest {
+
+  /*
+   *  GetOccupiedBytes tier statistic value is not updated after cache clearing
+   *  Scenario1 using DefaultStatisticsService
+   *
+   * */
+  @Test
+  public void testScenario1() throws StoreAccessException {
+
+    DefaultStatisticsService statisticsService = new DefaultStatisticsService();
+    CacheManager cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+      .using(statisticsService)
+      .build(true);
+
+    CacheConfiguration<String, String> cacheConfiguration = CacheConfigurationBuilder
+      .newCacheConfigurationBuilder(
+        String.class, String.class,
+        ResourcePoolsBuilder.newResourcePoolsBuilder().heap(100, MemoryUnit.KB)
+      ).withExpiry(ExpiryPolicyBuilder.timeToLiveExpiration(Duration.ofSeconds(100)))
+      .build();
+
+    Cache<String, String> cache = cacheManager.createCache("cacheName", cacheConfiguration);
+    cache.put("key1", "value1");
+    cache.put("key2", "value2");
+    cache.put("key3", "value3");
+
+    long occupiedByteSize = statisticsService.getCacheStatistics("cacheName").getTierStatistics().get("OnHeap").getOccupiedByteSize();
+    assertThat(occupiedByteSize, is(984L));
+
+    cache.clear();
+    occupiedByteSize = statisticsService.getCacheStatistics("cacheName").getTierStatistics().get("OnHeap").getOccupiedByteSize();
+    assertThat(occupiedByteSize, is(984L));
+    assertThat(occupiedByteSize, not(0L));
+
+    cache.remove("key1");
+    cache.remove("key2");
+    cache.remove("key3");
+    long currentMemoryPostRemove = statisticsService.getCacheStatistics("cacheName").getTierStatistics().get("OnHeap").getOccupiedByteSize();
+    assertThat(occupiedByteSize, is(984L));
```

Comment:
Don't you want to use `currentMemoryPostRemove` here?

Reply:
Wanted to remove this `currentMemoryPostRemove `variable usage by `occupiedByteSize`, corrected this now
I am simply trying to get the `getOccupiedByteSize()` post `remove()`